### PR TITLE
Jd/add ext

### PR DESCRIPTION
**Pull request by jd-lara**
This PR adds the additional constructor when ext is included in the structs definition. I added to the PR a change in the way internals is printed to enable dynamics.

### CODE REVIEW

File: src/utils/generate_structs.jl [hunk 1]
```diff
@@ -104,6 +114,16 @@ function generate_structs(directory, data::Vector; print_results=true)
                 continue
             end
 
+            if param["name"] == "ext"
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl]: You could make this an if/elseif/elseif block.

File: src/utils/generate_structs.jl [hunk 2]
```diff
@@ -53,6 +53,16 @@ function {{struct_name}}{T}({{#parameters}}{{^internal}}{{name}}{{#default}}={{d
     {{struct_name}}({{#parameters}}{{^internal}}{{name}}, {{/internal}}{{/parameters}}InfrastructureSystemsInternal())
 end
 {{/parametric}}
+
+{{#defines_ext}}
+function {{struct_name}}({{#parameters}}{{^internal}}{{^ext}}{{^_forecasts}}{{name}}, {{/_forecasts}}{{/ext}}{{/internal}}{{/parameters}}; ext={{#parameters}}{{#ext}}{{default}}{{/ext}}{{/parameters}})
+    {{#parameters}}
+    {{/parameters}}
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl]: Isn't this unnecessary?

File: src/utils/generate_structs.jl [hunk 3]
```diff
@@ -53,6 +53,16 @@ function {{struct_name}}{T}({{#parameters}}{{^internal}}{{name}}{{#default}}={{d
     {{struct_name}}({{#parameters}}{{^internal}}{{name}}, {{/internal}}{{/parameters}}InfrastructureSystemsInternal())
 end
 {{/parametric}}
+
+{{#defines_ext}}
+function {{struct_name}}({{#parameters}}{{^internal}}{{^ext}}{{^_forecasts}}{{name}}, {{/_forecasts}}{{/ext}}{{/internal}}{{/parameters}}; ext={{#parameters}}{{#ext}}{{default}}{{/ext}}{{/parameters}})
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl]: Why did you have to do something with forecasts?  Seems like the existing logic with 'default' should have taken care of it.

File: src/utils/generate_structs.jl (hunk 2 above)
Comment (jd-lara) [reply src/utils/generate_structs.jl]: I will test it, I based it from the code in line 51.

File: src/utils/generate_structs.jl (hunk 3 above)
Comment (daniel-thom) [reply src/utils/generate_structs.jl]: You could remove the `defines_ext` block if you define ext before _forecast in the descriptors.

Comment (jd-lara) [reply src/utils/generate_structs.jl]: I added this to avoid printing the _forecast argument.

File: src/generated/Deterministic.jl [hunk 4]
```diff
@@ -27,6 +27,8 @@ function Deterministic(; label, data, )
 end
 
 
+
```
Comment (daniel-thom) [inline src/generated/Deterministic.jl]: There are a bunch of extra blank lines from something in the Mustache template.

File: src/utils/generate_structs.jl [hunk 5]
```diff
@@ -104,6 +96,15 @@ function generate_structs(directory, data::Vector; print_results=true)
...
 
+            if param["name"] == "ext"
+                param["ext"] = true
+                continue
+            end
+
+            if param["name"] == "_forecasts"
+                param["_forecasts"] = true
+            end
+
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl]: This is no longer needed, right?

File: src/utils/generate_structs.jl [hunk 6]
```diff
@@ -35,21 +35,14 @@ mutable struct {{struct_name}}{{#parametric}}{T <: {{parametric}}}{{/parametric}
     end
     {{/inner_constructor_check}}
 end
-
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl]: Did you mean to remove these blank lines?  Now the functions in the generated code have no whitespace between them.

**Pull request merged**